### ISSUE TITLE
angel3_orm MySQL/MariaDB auto increment

### ISSUE THREAD

**Issue by Garthi** [dart-backend/angel#122]
Hello, we are currently searching for an ORM plugin for our new DART project and we tested the angel3_orm. But we can't find an auto increment definition for the entity.

Does it really not exist in your ORM or did we just not find the correct way to implement it? Could you give me some feedback regarding this matter? Thank you!

**Comment by dukefirehawk** [comment]
Have you tried the following?

@primayKey
int? id;

Take a look at the examples in packages/orm/angel_orm_test.

**Comment by Garthi** [comment]
Hi, we found two solutions one is 

```
  @Column(type: ColumnType.serial, indexType: IndexType.primaryKey)
  int? get id;
```
but with this version you can't add references, then the type alias generates a big int
the second solution is
```
  @Column(type: ColumnType.int, indexType: IndexType.primaryKey, length: 10, defaultValue: 'NULL AUTO_INCREMENT')
  int? get id;
```
this version is a bit hacki but is works for us :-)

**Comment by dukefirehawk** [comment]
The first solution should work. Will schedule it to be fixed.

**Comment by dukefirehawk** [comment]
This is fixed in `angel3_migration_runner` 8.2.1. integer field primary key will have `not null auto_increment` added during table creation. Serial field primary key will remain as it is. In MySQL, it is shorthand for `bigint not null auto_increment`, whereas in PostgreSQL it is `int not null default nextval(<tablename_sequence>)`. So in MySQL, to use serial as the primary key, would need foreign key to be bigint too.
